### PR TITLE
Add quat:ToEuler and quat:fromEuler

### PR DESCRIPTION
**Pull request by xiejiangzhi**

### CODE REVIEW

**Comment by bjornbytes** [comment]
I don't know much about euler angles.  Is ZXY the most common order to use for the different angles?

**Comment by xiejiangzhi** [comment]
> I don't know much about euler angles. Is ZXY the most common order to use for the different angles?

Me too. I’ve read some codes and the most commonly used ones are zyx
But I just used it and found that in the game, the values of pitch and yaw are reversed.

**Comment by xiejiangzhi** [comment]
After some testing, it was discovered that this conversion had a bug. . .
Because I don’t understand this conversion, the code is modified after copying.

**Comment by xiejiangzhi** [comment]
OK, I have fixed the bug, 
Now, it's result is the same as (three.js math) https://www.andre-gaschler.com/rotationconverter/  
Rotation order is YXZ

**Comment by bjornbytes** [comment]
I'd like to merge this, but can we rename the functions to get/set prefix instead of from/to, to match the naming of other methods?

- get/setEuler
- get/setAngles
- get/setEulerAngles

Also, the math should be done with floats instead of doubles, to avoid conversions (sinf/cosf/atan2f instead of sin/cos/atan2, and .5f instead of .5).  I can do this after merging though.

**Comment by xiejiangzhi** [comment]
It has been modified as you said

**Comment by bjornbytes** [comment]
Thanks!  And sorry about the delay

File: src/api/l_math_vectors.c [hunk 1]
```diff
@@ -1574,6 +1594,8 @@ const luaL_Reg lovrQuat[] = {
   { "direction", l_lovrQuatDirection },
   { "conjugate", l_lovrQuatConjugate },
   { "slerp", l_lovrQuatSlerp },
+  { "getEuler", l_lovrQuatGetEuler },
+  { "SetEuler", l_lovrQuatSetEuler },
```
Comment (jmiskovic) [inline src/api/l_math_vectors.c:1598]: Should be `setEuler`

**Pull request merged**